### PR TITLE
Add feature column compilation

### PR DESCRIPTION
**Pull request by sneaxiy**
Add `python/runtime/feature/compile.py` to compile the IR feature column map to runtime feature column map (i.e., call the TensorFlow or XGBoost feature column APIs). It is similar to the Go function `GenerateFeatureColumnCode`, but it returns the runtime feature column object instead of the string code.

### CODE REVIEW

File: python/runtime/feature/compile.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                raise ValueError(
+                    "field in CROSS must be of FeatureColumn or string type")
+
+        return fc_package.crossed_column(key_strs,
+                                         hash_bucket_size=fc.hash_bucket_size)
+
+    if isinstance(fc, EmbeddingColumn):
+        assert model_type != EstimatorType.XGBOOST, \
+            "EMBEDDING is not supported in XGBoost models"
+        cc = compile_feature_column(fc.category_column, model_type, package)
```
Comment (brightcoder01) [inline python/runtime/feature/compile.py]: What does cc stand for?

File: python/runtime/feature/compile.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    raise ValueError("unsupported data type {}".format(dtype))
+
+
+def compile_feature_column(fc, model_type, package):
+    """
+    Compile a IR FeatureColumn object to a runtime feature column object.
+
+    Args:
+        fc (FeatureColumn): the IR FeatureColumn object.
+        model_type (EstimatorType): one of TENSORFLOW and XGBOOST.
```
Comment (brightcoder01) [inline python/runtime/feature/compile.py:60]: model_type -> engine_type？

File: python/runtime/feature/compile.py [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+from runtime.model import EstimatorType
+
+__all__ = [
+    'compile_ir_feature_columns',
+]
+
+
+def to_package_dtype(dtype, package):
+    """
+    Convert dtype to the data type accepted by package.
```
Comment (typhoonzero) [inline python/runtime/feature/compile.py]: by package -> by the feature column implementation packages including `tensorflow` and `xgboost`.

File: python/runtime/feature/compile.py [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+    Args:
+        dtype (DataType): one of INT, FLOAT and STRING.
+        package (module): the Python package.
+
+    Returns:
+        The data type accepted by the package.
+    """
+    if dtype == DataType.INT:
+        return package.dtypes.int64
```
Comment (typhoonzero) [inline python/runtime/feature/compile.py:43]: Can we have a universal package dtype?

File: python/runtime/feature/compile.py [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+    raise ValueError("unsupported data type {}".format(dtype))
+
+
+def compile_feature_column(fc, model_type, package):
+    """
+    Compile a IR FeatureColumn object to a runtime feature column object.
+
+    Args:
+        fc (FeatureColumn): the IR FeatureColumn object.
```
Comment (brightcoder01) [inline python/runtime/feature/compile.py]: How about marking fc specifically to `fc_ir`

File: python/runtime/feature/compile.py (hunk 3 above)
Comment (sneaxiy) [reply python/runtime/feature/compile.py]: Done.

File: python/runtime/feature/compile.py (hunk 4 above)
Comment (sneaxiy) [reply python/runtime/feature/compile.py:43]: A bit hard. 

- If we have a universal package dtype and it depends on a framework (say, in the form of `tf.dtypes.xxx`), we have to convert it to XGBoost-compatible dtype when training the XGBoost model. 
- If we have a universal package dtype that is independent of any framework (the current implementation), we have to convert it to the framework related dtype when training.

In a word, it seems that the conversation is inevitable no matter whether we have a universal package dtype.

File: python/runtime/feature/compile.py (hunk 1 above)
Comment (sneaxiy) [reply python/runtime/feature/compile.py]: I have replaced `cc` with `category_column` for a better review.

File: python/runtime/feature/compile.py (hunk 5 above)
Comment (sneaxiy) [reply python/runtime/feature/compile.py]: Done. Renamed to `ir_fc`.

File: python/runtime/feature/compile.py (hunk 2 above)
Comment (sneaxiy) [reply python/runtime/feature/compile.py:60]: To be compatible with the naming rules in https://github.com/sql-machine-learning/sqlflow/blob/develop/python/runtime/pai/submitter.py#L518 , I prefer to use `model_type` here.

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Review by brightcoder01 — APPROVED** [review]

**Pull request merged**